x-pack/plugins/secops/public/components/page/network/ip_overview/field_renderers.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { getOr, isString } from 'lodash/fp';
+import React from 'react';
+
+import {
+  AutonomousSystem,
+  HostEcsFields,
+  IpOverviewData,
+  IpOverviewType,
+  Overview,
+} from '../../../../graphql/types';
+import { DefaultDraggable } from '../../../draggables';
+import { getEmptyTagValue } from '../../../empty_value';
+import { ExternalLinkIcon } from '../../../external_link_icon';
+import { PreferenceFormattedDate } from '../../../formatted_date';
+import { HostDetailsLink, ReputationLink, VirusTotalLink, WhoIsLink } from '../../../links';
+
+import { IpOverviewId } from './index';
+import * as i18n from './translations';
+
+export const locationRenderer = (
+  fieldNames: string[],
+  data: IpOverviewData
+): React.ReactElement => {
+  return fieldNames.length > 0 &&
+    fieldNames.every((fieldName: string) => getOr(null, fieldName, data)) ? (
+    <EuiFlexGroup alignItems="center" gutterSize="none" data-test-subj="location-field">
+      {fieldNames.map((fieldName: string, index: number) => {
```

Comment:
fwiw, you don't need to type out the values of `map` or `every` or other array types as TypeScript will know what the values are automatically.

So you can do:

```ts
fieldNames.map((fieldName, index)
```

and 

```ts
fieldNames.every(fieldName => getOr(null, fieldName, data))
```

If you want to.

Reply:
Ahh, thanks for the clarification on the type inferencing. I think I was operating from a previous scenario where I had done a:

``` javascript
const summaryValue: string | string[] | null = get(field, host);
```
and then needed to explicitly type like the following because of the `getOr`:
``` javascript 
summaryValue.map((value: string) => {});
```
Went ahead cleaned this up, thanks!